hepcrawl/parsers/osti.py

```diff
@@ -0,0 +1,362 @@
+
+from lxml.etree import XMLSyntaxError
+
+
+class OSTIParser(object):
+    """Parser for the OSTI json format.
+    """
+    def __init__(self, record, source=u'OSTI'):
+        self.record = record
+        self.builder = LiteratureBuilder(source)
+
+    def parse(self):
+        """Extract a list of OSTI json records into INPIRE HEP records.
+
+        Returns:
+            generator of dicts: records in the Inspire Literature schema.
+        """
+        self.builder.add_abstract(self.abstract)
+        for author in self.authors:
+            self.builder.add_author(author)
+        self.builder.add_document_type(self.document_type)
+        for collection in self.collections:
+            self.builder.add_collection(collection)
+        for collaboration in self.collaborations:
+            self.builder.add_collaboration(collaboration)
+        for doi in self.dois:
+            self.builder.add_doi(**doi)
+        # for keyword in self.keywords:
+        #     self.builder.add_keyword(**keyword)
+        self.builder.add_external_system_identifier(self.osti_id, self.source)
+        self.builder.add_imprint_date(self.date_published)
+        self.builder.add_publication_info(**self.publication_info)
+        for report_number in self.report_numbers:
+            self.builder.add_report_number(report_number, self.source)
+        self.builder.add_title(self.title)
+
+        return self.builder.record
+
+    @property
+    def abstract(self):
+        """Abstract/description of the publication.
+
+        Returns:
+            str:
+        """
+        abstract = self.record.get(u'description')
+        if not abstract:
+            return None
+        remove_tags_config_abstract = {
+            'allowed_tags': [u'sup', u'sub'],
+            'allowed_trees': [u'math'],
+            'strip': 'self::pub-id|self::issn'
+        }
+        try:
+            abstract = remove_tags(abstract, **remove_tags_config_abstract).strip()
+        except XMLSyntaxError:
+            pass
+
+        return abstract
+
+    @property
+    def authors(self):
+        """Authors and affiliations.
+
+        Returns: list of author dicts
+
+        """
+        authors, _ = self._get_authors_and_affiliations(
+            self.record.get(u'authors'))
+        parsed_authors = [self.builder.make_author(
+            full_name=author.get(u'full_name'),
+            raw_affiliations=author.get(u'affiliations'),
+            ids=[('ORCID', author.get(u'orcid'))])
+                          for author in authors]
+        return parsed_authors
+
+    @property
+    def collaborations(self):
+        """Participating collaborations.
+
+        Returns:
+            list: of str, e.g. [u'VERITAS', u'Fermi-LAT']
+        """
+        return [t[4:] if t.startswith(u'The ') else t for t in
+                [c.replace(u'Collaboration', '').strip() for c in
+                 self.record.get(u'contributing_org', '').split(u';')]]
```

Comment:
agreed. _RE_AND could be augmented to split on `;` in addition.
however on things like 
`'LUX Collaboration; Nuclear Science and Security Consortium'`
the string should not be split on `and` 

it's the only outlier I see in 3k records I just checked, though.

another problem I see with splitting at `;` in general is that it might mess up HTML escapes like `&amp;`  Is there a step to clean those before splitting? are you going to augment schema utils?

Reply:
There is definitely mixed content with `;` and `and` as separator
`The DES Collaboration; The LIGO Scientific Collaboration and the Virgo Collaboration`

Reply:
There is also repetition after splitting and removing things, e.g. in
```
LIGO Scientific Collaboration; Virgo Collaboration; Fermi GBM; INTEGRAL; IceCube Collaboration; AstroSat Cadmium Zinc Telluride Imager Team; IPN Collaboration; The Insight-Hxmt Collaboration; ANTARES Collaboration; The Swift Collaboration; AGILE Team; The 1M2H Team; The Dark Energy Camera GW-EM Collaboration; the DES Collaboration; The DLT40 Collaboration; LIGO Scientific Collaboration and Virgo Collaboration; The Insight-HXMT Collaboration; The Dark Energy Camera GW-EM Collaboration and the DES Collaboration
```
are the schema/utils deduping the list ?

Reply:
> another problem I see with splitting at `;` in general is that it might mess up HTML escapes like `&amp;` Is there a step to clean those before splitting? are you going to augment schema utils?

I don't think dealing with various text encodings and escaping schemes should be part of the schema utils honestly. The crawler should know what format it expects and convert to unescaped unicode.


> are the schema/utils deduping the list ?

Not currently, but it could be added (there's utils for it in `inspire_utils.dedupers`).